users/views.py

```diff
@@ -70,4 +72,62 @@ def post(self, *args, **kwargs):
                                               password=self.request.POST['password'])
             login(self.request, user)
             return redirect('dashboard')
-        return render(self.request, self.template_name, {'form':form})
\ No newline at end of file
+        return render(self.request, self.template_name, {'form':form})
+
+
+class UserProfileView(View):
+    """ display a user's profile
+    """
+    template_name = 'user/profile.html'
+
+    def get(self, *args, **kwargs):
+        """ render a user's profile
+        """
+        # get the user that is currently logged in
+        return render(self.request, self.template_name, {})
+
+
+class UpdateProfileView(TemplateView):
+    """ Update the user's first and last names
+    """
+    template_name = 'user/edit.html'
+
+    def get(self, *args, **kwargs):
+        """ display
+        """
+        form = UpdateProfileModelForm(instance=self.request.user)
+        return render(self.request, self.template_name, {'form':form})
+
+    def post(self, *args, **kwargs):
+        """update the user's profile (first name, last name)
+        """
+        form = UpdateProfileModelForm(self.request.POST,instance=self.request.user)
```

Comment:
Don't name the form ModelForm. all of them are `Form`. just <Name>Form format is enough